auslib/db.py

```diff
@@ -1085,7 +1088,38 @@ def updateRelease(self, name, changed_by, old_data_version, product=None, read_o
             if self.containsForbiddenDomain(blob, product):
                 raise ValueError("Release blob contains forbidden domain.")
             what['data'] = blob.getJSON()
-        self.update(where=[self.name == name], what=what, changed_by=changed_by, old_data_version=old_data_version, transaction=transaction)
+        try:
+            self.update(where=[self.name == name], what=what, changed_by=changed_by, old_data_version=old_data_version, transaction=transaction)
+        except OutdatedDataError as e:
+            if blob is not None and read_only is None:
+                ancestor_blob = createBlob(self.history.getChange(data_version=old_data_version,
+                                                                  column_values={'name': name},
+                                                                  transaction=transaction)
+                                                       .get('data'))
```

Comment:
Style nit: Assigning the getChange result and then passing it to createBlob would probably be much cleaner. The inline call to .get() after a multiple line invocation is tough to read.